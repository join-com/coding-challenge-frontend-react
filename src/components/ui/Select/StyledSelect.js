import styled from "styled-components";
import { defaultInputStyles } from "../../../helpers/stylesMixins";
import { defaultIndent } from "../../../constants/styles";
import StyledText from "../Text/StyledText";

export const StyledSelect = styled.select`
  ${defaultInputStyles};
`;

export const Label = styled(StyledText)`
  padding-right: ${defaultIndent}rem;
`;
