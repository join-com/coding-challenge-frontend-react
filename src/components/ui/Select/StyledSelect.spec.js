import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import { StyledSelect, Label } from "./StyledSelect";

describe("StyledSelect component", () => {
  const defaultProps = {
    disabled: false
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return renderer.create(<StyledSelect {...props} />).toJSON();
  };

  it("should match snapshot ", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot for disabled prop ", () => {
    const wrapper = getWrapper({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Styled Label component", () => {
  it("should match snapshot ", () => {
    expect(renderer.create(<Label />).toJSON()).toMatchSnapshot();
  });
});
