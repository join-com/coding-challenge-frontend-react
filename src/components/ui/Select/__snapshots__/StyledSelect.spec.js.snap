// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Label component should match snapshot  1`] = `
.c0 {
  position: relative;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: left;
  color: #1D232B;
  text-transform: none;
  margin: 0 0 0.5rem 0;
  padding-right: 0.5rem;
}

<p
  className="c0"
/>
`;

exports[`StyledSelect component should match snapshot  1`] = `
.c0 {
  border: solid 2px #1D232B;
  border-radius: 3px;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2;
  background-color: #F9F9F9;
  color: #1D232B;
  opacity: 1;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  border-radius: 3px;
  text-transform: none;
  cursor: pointer;
}

.c0:hover {
  border-color: #90D3DC;
}

<select
  className="c0"
  disabled={false}
/>
`;

exports[`StyledSelect component should match snapshot for disabled prop  1`] = `
.c0 {
  border: solid 2px #1D232B;
  border-radius: 3px;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2;
  background-color: #F9F9F9;
  color: #1D232B;
  opacity: 0.5;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  border-radius: 3px;
  text-transform: none;
  cursor: not-allowed;
}

<select
  className="c0"
  disabled={true}
/>
`;
